api/src/main/java/jakarta/persistence/criteria/TemporalField.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2008, 2020 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+package jakarta.persistence.criteria;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+
+/**
+ * Each instance represents a type of field which can be extracted from a
+ * date, time, or datetime.
+ * @param <T> the resulting type of the extracted value
+ * @see CriteriaBuilder#extract(TemporalField, Expression)
+ */
+public class TemporalField<T> {
```

Comment:
This would need to be an interface.

Reply:
Either that, or make the constructor `protected`.